azkaban-web-server/src/main/java/azkaban/flowtrigger/quartz/FlowTriggerScheduler.java

```diff
@@ -163,6 +163,14 @@ public boolean resumeFlowTriggerIfPresent(final int projectId, final String flow
               flowId, flowTrigger, submitUser, quartzTriggers.isEmpty() ? null
               : quartzTriggers.get(0), isPaused, flow.isLocked());
         } catch (final Exception ex) {
+          if (ex instanceof org.quartz.JobPersistenceException) {
+            if (ex.getCause() instanceof java.io.InvalidClassException) {
+              if(this.scheduler.enableSerializationHack()) {
+                logger.info("Enable serialization hack");
+                return getScheduledFlowTriggerJobs();
```

Comment:
I must admit though, the more I look into this, the more I respect what you've accomplished here... a true hack.

For my education: why can't we directly extend `StdJDBCDelegate`, overwrite `getObjectFromBlob` *only* with your filtering version, tell Quartz (via `org.quartz.jobStore.driverDelegateClass` configuration property) to use that delegate from get-go ...and call it good? 

Extra benefit is that anyone who hasn't been exposed to the new UID, or who purged their DB of it, either deliberately or over time, doesn't need to bother and can just use straight `StdJDBCDelegate` again.

Reply:
Yes, one of the scenarios where it will return false is if the targeted inner class is always of the wrapped type. This function will not re-hack the scheduler; it will return false.

Reply:
I've also made the exception recognition much more specific, found now in CustomQuartzDeserializer.isSerializationBug